### PR TITLE
[GSoC] Kotlin migration Anki2Importer.java

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Kotlin Migration

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt [hunk 1]
```diff
@@ -515,390 +488,403 @@ public class Anki2Importer extends Importer {
...
      * @param notesDone Percentage of notes complete.
      * @param cardsDone Percentage of cards complete.
      * @param postProcess Percentage of remaining tasks complete.
      */
-    protected void publishProgress(int notesDone, int cardsDone, int postProcess) {
+    protected fun publishProgress(notesDone: Int, cardsDone: Int, postProcess: Int) {
         if (mProgress != null) {
-            mProgress.publishProgress(getRes().getString(R.string.import_progress,
-                    notesDone, cardsDone, postProcess));
+            mProgress.publishProgress(
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:869]: cleanup: Could simplify with `?.`

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt [hunk 2]
```diff
@@ -515,390 +488,404 @@ public class Anki2Importer extends Importer {
...
-    }
-
-
-    public int getUpdated() {
-        return mUpdated;
+    init {
+        mNeedMapper = false
+        mDeckPrefix = null
+        mAllowUpdate = true
+        mDupeOnSchemaChange = false
```
Comment (divyansh-dxn) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:889]: cleanup: combined declaration and initialization.
@lukstbit agree?

Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/libanki/importer/Anki2Importer.kt:889]: Yes, I forgot about that.

**Review by lukstbit — APPROVED** [review]

**Review by david-allison — APPROVED** [review]
That looked like a difficult one, thanks!

**Pull request merged**